src/control/nav2_controller_example/include/nav2_controller_example/dwa_controller.hpp

```diff
@@ -17,17 +17,21 @@
 
 #include <string>
 #include "nav2_tasks/follow_path_task.hpp"
+#include "geometry_msgs/msg/twist.hpp"
 
 namespace nav2_controller_example
 {
 
 class DwaController : public nav2_tasks::FollowPathTaskServer
 {
 public:
+  typedef geometry_msgs::msg::Twist CmdVel;
   DwaController();
   ~DwaController();
 
+protected:
   nav2_tasks::TaskStatus execute(const nav2_tasks::FollowPathCommand::SharedPtr path) override;
+  std::shared_ptr<rclcpp::Publisher<CmdVel>> vel_pub_;
```

Comment:
unique_ptr instead?

Reply:
That would make sense, but the `create_publisher` function returns a `shared_ptr` and it seems there is no convenient way to convert from `shared` to `unique`. The other direction is fine though.